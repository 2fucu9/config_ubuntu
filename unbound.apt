#prep
lsof -i -P -n | grep LISTEN

mkdir -p /etc/systemd/resolved.conf.d
nano /etc/systemd/resolved.conf.d/adguardhome.conf

[Resolve]
DNS=127.0.0.1
DNSStubListener=no

mv /etc/resolv.conf /etc/resolv.conf.backup
ln -s /run/systemd/resolve/resolv.conf /etc/resolv.conf
#systemctl reload-or-restart systemd-resolved
reboot

lsof -i -P -n | grep LISTEN

#https://www.techrepublic.com/article/how-to-install-unbound-dns/
apt-get install unbound -y
reboot

nano /etc/unbound/unbound.conf.d/myunbound.conf

<#
server:
  interface: 127.0.0.1
  interface: ::0
  access-control: 10.14.0.0/16 allow
  access-control: ::1 allow
  verbosity: 1
  port: 5353
  do-ip6: no
  do-ip4: yes
  do-udp: yes
  do-tcp: yes
  # Set number of threads to use
  num-threads: 4
  # Hide DNS Server info
  hide-identity: yes
  hide-version: yes
  # Limit DNS Fraud and use DNSSEC
  harden-glue: yes
  harden-dnssec-stripped: yes
  harden-referral-path: yes
  use-caps-for-id: yes
  harden-algo-downgrade: no
  qname-minimisation: yes
  aggressive-nsec: yes
  rrset-roundrobin: yes

  # If DNSSEC isnt working uncomment the following line
  # auto-trust-anchor-file: "/var/lib/unbound/root.key"


  # Minimum lifetime of cache entries in seconds
  cache-min-ttl: 300
  # Configure TTL of Cache
  cache-max-ttl: 14400
  # Optimizations
  msg-cache-slabs: 8
  rrset-cache-slabs: 8
  infra-cache-slabs: 8
  key-cache-slabs: 8
  serve-expired: yes
  serve-expired-ttl: 3600
  edns-buffer-size: 1232
  prefetch: yes
  prefetch-key: yes
  target-fetch-policy: "3 2 1 1 1"
  unwanted-reply-threshold: 10000000
  # Set cache size
  rrset-cache-size: 256m
  msg-cache-size: 128m
  # increase buffer size so that no messages are lost in traffic spikes
  so-rcvbuf: 1m
  private-address: 192.168.0.0/16
  private-address: 169.254.0.0/16
  private-address: 172.16.0.0/12
  private-address: 10.0.0.0/8
  private-address: fd00::/8
  private-address: fe80::/10

#>

touch /var/log/unbound.log
chown unbound:unbound /var/log/unbound.log
systemctl enable --now unbound
reboot
dig google.com @localhost -p 5353









