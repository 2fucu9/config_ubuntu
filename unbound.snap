apt-get install unbound -y
